### DISCUSSION TITLE
Using primitives or Objects

### DISCUSSION THREAD

**Discussion by sdelamo** [micronaut-projects/micronaut-guides#810] [General]
@burtbeckwith PRs are replacing the [usage of objects with primitives](https://github.com/micronaut-projects/micronaut-guides/pull/803). I personally prefer using Objects instead primitives but we should decide what do as a guide style and be consistent. 

Please note that our documentation, for example in [Micronaut Data](https://micronaut-projects.github.io/micronaut-data/latest/guide/), uses `Objects` instead of primitives

```java
@Entity
public class Book {
    @Id
    @GeneratedValue
    private Long id;
```

```java
   @NonNull
    @Override
    Mono<Author> findById(@NonNull @NotNull Long aLong);
```

Not primitives

```java
@Entity
public class Book {
    @Id
    @GeneratedValue
    private long id;
```

```java
   @NonNull
    @Override
    Mono<Author> findById(long aLong);
```

@graemerocher @shingjo @alvarosanchez @graemerocher @burtbeckwith thoughts?

**Comment by alvarosanchez** [comment]
I don't have a strong opinion here. I also tend to prefer to use primitives unless their default value might change semantics (e.g. `0` or `false` vs `null`).

**Comment by burtbeckwith** [comment]
To clarify, I've been replacing `@NonNull @NotNull Long` with `long` and `@NonNull @NotNull Integer` with `int`. If you don't want the number to be null, use a primitive.

This approach can cause issues with detecting whether a value has been set since a primitive numeric field is initialized to 0 and it makes it hard to know if the 0 value was set, or if it was just the initialized value. But that seems to be an issue in JPA/Hibernate and not Micronaut Data JDBC.

**Comment by shingjo** [comment]
Consistency is nice; however, can give the impression that it is the only way. 

Main reason I would argue for use of both in the guides is that they are *under test*. 

IMHO: Having `@NonNull @NotNull` in addition to whatever other validation is needed clutters up the code. Plus, they are so similar that I still get them confused.
